### ISSUE TITLE
enhance behavior of ReferenceEquality

### ISSUE THREAD

**Issue by regrog** [google/error-prone#440]
Is there any chance to enhance ReferenceEquality?
many people use reference equality to check enums.
In my case  I have this warning where Level is an enum.

Thnks

```
    [javac] WindowAppender.java:34: warning: [ReferenceEquality] Comparison using reference equality instead of value equality
    [javac]         if (event.getLevel() == Level.ERROR) {
    [javac]                              ^
    [javac]     (see http://errorprone.info/bugpattern/ReferenceEquality)
    [javac]   Did you mean 'if (Objects.equals(event.getLevel(), Level.ERROR)) {' or 'if (event.getLevel().equals(Level.ERROR)) {'?
```

**Comment by cushon** [comment]
It shouldn't flag reference equality on enums: https://github.com/google/error-prone/blob/596e701bf46cda55f1069e13926fa281e73c142f/core/src/main/java/com/google/errorprone/bugpatterns/ReferenceEquality.java#L63

What does the definition of `Level.ERROR` look like? Do you have a repro?

**Comment by regrog** [comment]
Sorry it's my fault, Level is a log4j Class and it's not an enum ....
